_studio/shared/umc/codec/vp9_dec/src/umc_vp9_va_packer.cpp

```diff
@@ -105,6 +105,36 @@ void PackerVA::PackAU(VP9Bitstream* bs, VP9DecoderFrame const* info)
 
     std::copy(data + offset, data + offset + length, bistreamData);
     pCompBuf->SetDataSize(length);
+
+    if(m_va->m_MaxContextPriority)
+        PackPriorityParams();
+}
+
+void PackerVA::PackPriorityParams()
+{
+    mfxPriority priority = m_va->m_ContextPriority;
+    UMCVACompBuffer *GpuPriorityBuf;
+    VAContextParameterUpdateBuffer* GpuPriorityBuf_Vp9Decode = (VAContextParameterUpdateBuffer *)m_va->GetCompBuffer(VAContextParameterUpdateBufferType, &GpuPriorityBuf, sizeof(VAContextParameterUpdateBuffer));
+    if (!GpuPriorityBuf_Vp9Decode)
+        throw vp9_exception(MFX_ERR_MEMORY_ALLOC);;
+
+    memset(GpuPriorityBuf_Vp9Decode, 0, sizeof(VAContextParameterUpdateBuffer));
+    GpuPriorityBuf_Vp9Decode->flags.bits.context_priority_update = 1;
+
+    if(priority == MFX_PRIORITY_LOW)
+    {
+        GpuPriorityBuf_Vp9Decode->context_priority.bits.priority = 0;
+    }
+    else if (priority == MFX_PRIORITY_HIGH)
+    {
+        GpuPriorityBuf_Vp9Decode->context_priority.bits.priority = m_va->m_MaxContextPriority;
+    }
+    else
+    {
+    GpuPriorityBuf_Vp9Decode->context_priority.bits.priority = m_va->m_MaxContextPriority/2;
```

Comment:
So MSDK only supports 3 levels. But media driver may support more levels. May we need expand it in the future?
Another question: Should we add it for AV1 decoding too? I am not sure.

Reply:
These are questions to @hongbolv  and @dylanintel . Folks, can you please reply?

Reply:
@wangyan-intel  In the beginning, we design 3 levels to align with graphics side. About the AV1 decoding, we should add it.

Reply:
Another question: The code section seems similar in every codec. So could we use common function for avoiding code duplication? Thanks.

Reply:
@wangyan-intel , It's a very good question. You know, when I reviewed the PR the first time my desire was to propose to decouple the change into some separate code/class. But later decided not to raise/insist on concern because it may require quite a big refactoring which I'm not sure Dylan is ready to make. Ideal solution for me would be to move this GPU priority logic out of codec level code, e.g. to move it to a wrapper under libva level and switch each component to this wrapper instead of libva.  @dylanintel , @hongbolv, please feel free to comment.

Reply:
Frankly speaking, I am thinking about this issue. The requirement is that priority setting per frame, then we have to pass the priority value through vaRenderPicture calling on current implementation. It made us to change every component because there is no a common libva wrapper in the media SDK. If we plan to add a libva wrapper for each component, i think it is a big refactoring effort too.